contracts/EllipticCurve.sol

```diff
@@ -251,6 +252,316 @@ contract EllipticCurve {
+    e = mulmod(e, b, p);  // f = b^3  (we will re-use the variable e )
+    uint256 temp = addmod(mulmod(t1, t1, p), p-addmod(mulmod(2, t0, p), e, p), p);
+    P[0] = temp;
+    temp = mulmod(t1, addmod(t0, p-temp, p), p);
+    P[1] = addmod(temp, p-mulmod(c, e, p), p);
+    P[2] = mulmod(b, mulmod(Pz, Qz, p), p);
+  }
+
+  /// @notice Subtraction of two points in Jacobian coordinates, placing the result in the first point
+  /// @dev Based on the addition formulas from http://www.hyperelliptic.org/EFD/g1p/auto-code/shortw/jacobian-0/addition/add-2001-b.op3
+  /// @param P An EC point in Jacobian coordinates. The result is returned here.
+  /// @param Q An EC point in Jacobian coordinates
+  function subJacobianMutates(
+    uint[3] memory P, 
+    uint[3] memory Q
+  )  internal pure {
+    uint256 Pz = P[2];
+    uint256 Qz = Q[2];
+    uint256 p = fieldOrder;
+
+    if (Pz == 0) {
+      P[0] = Q[0];
+      P[1] = p - Q[1];
+      P[2] = Qz;
+      return;
+    } else if (Qz == 0) {
+      return;
+    }
+
+    uint256 zz = mulmod(Pz, Pz, p);
+    uint256 t0 = mulmod(Q[0], zz, p);
+    uint256 t1 = mulmod(p - Q[1], mulmod(Pz, zz, p), p);
+
+    zz = mulmod(Qz, Qz, p);
+    uint256 a = mulmod(P[0], zz, p);
+    uint256 c = mulmod(P[1], mulmod(Qz, zz, p), p); 
+
+    if ((a == t0) && (c == t1)) {
+      P[2] = 0;
+      return;
+    }
+        
+    t1 = addmod(t1, p-c, p); // d = t1 - c
+    uint256 b = addmod(t0, p-a, p); // b = t0 - a
+    uint256 e = mulmod(b, b, p); // e = b^2
+    t0 = mulmod(a, e, p);    // t0 is actually "g"
+    e = mulmod(e, b, p);  // f = b^3  (we will re-use the variable e )
+    uint256 temp = addmod(mulmod(t1, t1, p), p-addmod(mulmod(2, t0, p), e, p), p);
+    P[0] = temp;
+    temp = mulmod(t1, addmod(t0, p-temp, p), p);
+    P[1] = addmod(temp, p-mulmod(c, e, p), p);
+    P[2] = mulmod(b, mulmod(Pz, Qz, p), p);
+  }
+
+  function doubleMutates(uint[3] memory P) internal pure {
+    uint256 z = P[2];
+    if (z == 0)
+      return;
+    uint256 p = fieldOrder;
+    uint256 x = P[0];
+    uint256 _2y = mulmod(2, P[1], p);
+    uint256 _4yy = mulmod(_2y, _2y, p);
+    uint256 s = mulmod(_4yy, x, p);
+    uint256 m = mulmod(3, mulmod(x, x, p), p);
+    uint256 t = addmod(mulmod(m, m, p), mulmod(0xfffffffffffffffffffffffffffffffffffffffffffffffffffffffefffffc2d, s, p),p);
+    P[0] = t;
+    P[1] = addmod(mulmod(m, addmod(s, p - t, p), p), mulmod(0x7fffffffffffffffffffffffffffffffffffffffffffffffffffffff7ffffe17, mulmod(_4yy, _4yy, p), p), p);
+    P[2] = mulmod(_2y, z, p);
+  }
+  function _lookup_sim_mul(
+    uint256[3][4][4] memory iP, 
+    uint256[4] memory P_Q,
+    uint256 a,
+    uint256 pp
+  ) internal pure {
+    uint256 p = fieldOrder;
+    uint256 beta = 0x7ae96a2b657c07106e64479eac3434e99cf0497512f58995c1396c28719501ee;
+    
+    uint256[3][4] memory iPj;
+    uint256[3] memory double;
+
+    // P1 Lookup Table
+    iPj = iP[0];
+    iPj[0] = [P_Q[0], P_Q[1], 1];  						// P1
```

Comment:
The comment is too far on the right side.

Reply:
This part of the code was removed :)